File: Project1_4.py
import cv2
import numpy as np
import datetime


# Frame'leri okunacak olan videoyuyu opencv kütüphanesi ile açıyoruz.
cap = cv2.VideoCapture("NvidiaJetsonNanoRobot.mp4") 

# Üzerinde işlem yapılan frame'ler tekrar kayıt edileceği için kayıt edilecek videonun pencere boyutları ve hızı ayarlanmaktadır.
frameRate = cap.get(cv2.CAP_PROP_FPS)
writer_width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
writer_height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))

# Kayıt edilecek video formatı ve yeni videonun ismi belirtilmektedir.
writer = None
(W,H) = (None, None)
fourcc = cv2.VideoWriter_fourcc(*'H264')
writer = cv2.VideoWriter("NvidiaJetsonNanoRobotOUTPUT.mp4", fourcc, frameRate, (writer_width, writer_height), True)

# Video'nun sonuna kadar framelerin okunması icin sonsuz dongu icinde tek tek frameleri okuyoruz.
while True:
  # frameleri okuma asamasi
    (_, img) = cap.read()
    img_width = img.shape[1] 
    img_height = img.shape[0]
                                   
        
    # Okunan framelere tarih ve saat bilgisi datetime modulü ile alınmakta ve opencv fonksiyonları ile yazılmaktadır.
    cv2.putText(img, "DATE : " + str(datetime.datetime.now().date()), (10, 30), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 0, 0), 3)
    cv2.putText(img, "TIME : " + str(datetime.datetime.now().time()), (10, 80), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 0, 0), 3)
    
    # Okunan ve üzerinde işlem yapılan videoyuyu bilgisayar ekranında işlem bitene kadar izlemek için  imshow fonksiyonu kullanılıyor ve write ile video kayıt ediliyor.
    cv2.imshow("OUTPUT", img)
    writer.write(img)
			
    # Bu kısım framelerin tamamı bitmeden kapatmak istersek ESC tuşu ile videoyuyu kapatmak için yazılmıştır.
    k = cv2.waitKey(5) & 0xFF
    if k == 27:
        break

# Tum islemler bittikten sonra tum pencelerin duzgun bir sekilde kapanmasini saglamaktadir.
writer.release()
cv2.destroyAllWindows()

File: Project1_4.cpp
#include <opencv2/core.hpp> 
#include <opencv2/dnn.hpp>
#include <opencv2/videoio.hpp> 
#include <opencv2/highgui.hpp> 
#include <opencv2/imgproc.hpp> 
#include <iostream> 
#include <time.h> 
#include <ctime>

int main() 
{ 
    cv::VideoCapture cap; 
 
    cap.open("SimpleObjectTrackingCamera.mp4", cv::CAP_ANY); 

  
    // Video ciktisinin kaydedilmesi icin kayit formati ve ayarlamalari belirliyoruz.  
  
    std::string outputFile = "CPP_OUTPUT_SimpleObjectTrackingCamera.mp4";
    cv::VideoWriter writer; 
	  writer.open(outputFile, cv::VideoWriter::fourcc('H','2','6','4'), int(cap.get(cv::CAP_PROP_FPS)), cv::Size((int)cap.get(cv::CAP_PROP_FRAME_WIDTH),(int) cap.get(cv::CAP_PROP_FRAME_HEIGHT)));

    for (;;) 
    { 
        if (!cap.isOpened()) 
        { 
            std::cout << "Video Capture Fail" << std::endl; 
            break; 
        } 
        cv::Mat img; 
        cap >> img; 
        
        time_t su_anki_saat = time(NULL);

            
        // saat ve tarih degerini alip stringe cevirerek ekrana yazdiriyoruz.
        char* cevrilmis = ctime(&su_anki_saat);
        std::cout<<"Cevrilmis saat : "<<cevrilmis<<std::endl;

        cv::putText(img, "DATE & TIME : " + cevrilmis, cv::Point(left, top), 1, 2, cv::Scalar(color, 255, 255), 2, false);

        cv::imshow("DisplayWindow", img); 

        writer.write(img);
        int k = cv::waitKey(5) & 0xFF;
        if(k == 27)
        {
            break;
        }
        
    } 
    writer.release();
    cv::destroyAllWindows();
    return 0;
}
